ABM/Modules/Demographic_functions.jl

```diff
@@ -0,0 +1,765 @@
+
+        else
+            error("""The growth form is undefined, 
+                     please check species demography data""")
+        end
+    end
+
+    #//-------------------------------------------------------------------------------------------#
+    #% NEIGHBOURHOOD DISPERSAL
+    """ 
+    # Neighbourhood dispersal
+    Seeds, represented as seedlings, are dispersed from target tree. Seeds are assumed to land 
+    under the crown of the target tree. The crown size of the tree is computed based about its 
+    diameter at breast height following a simplified allometric relationship from SORTIE-NZ
+    (Kunstler et *al*., 2009).
+    ## References:
+    - Kunstler, G., Coomes, D.A., Canham, C.D., 2009. Size-dependence of growth and mortality 
+    influence the shade tolerance of trees in a lowland temperate rain forest. J. Ecol. 97, 685-695. 
+    ## Arguments:
+    - `model`: The AgentBasedModel object defining the current model. This object is usually 
+    created using `Agents.ABM()`.
+    - `seed_production::Int64`: Species-specific constant value used within a Poisson distribution
+    to determine the number of seeds produced by an agent.
+    - `ldd_disp_frac::Float64`: Proportion of seedlings that undergo long-distance dispersal and are
+    therefore unavailable for neighbourhood dispersal.
+    - `r_hgt::Int64`: Regeneration height in meters for selected species.
+    - `dbh::Float64`: Diameter at breast height in meters of tree.
+    - `cell_grain::Int64`: The size of an individual cell in the model (cell_grain x cell_grain).
+    - `shell_layers::Int64`: Maximum distance at which neighbours may be considered to possibly fall
+    under the crown.
+    - `pos::Tuple{Int64,Int64}`: Coordinates of tree/agent.
+    - `nhbs_id::Vector{Int64}`: Patch IDs for all neighbouring cells of tree/agent.
+    - `shad_h::Vector{Float64}`: Mean shade heights as calculated by `get_nhb_shade_height()` for 
+    all cells in model grid.
+    - `seedlings::Vector{Vector{Int64}}`: Number of seedlings for each species for every cell in the
+    model grid.
+    - `spec_ID::Int64`: Selected species ID.
+    """
+    function nhb_dispersal(
+        model,
+        seed_production::Int64,
+        ldd_disp_frac::Float64,
+        r_hgt::Int64,
+        dbh::Float64,
+        cell_grain::Int64,
+        shell_layers::Int64,
+        pos::Tuple{Int64,Int64},
+        nhbs_id::Vector{Int64},
+        shad_h::Vector{Float64},
+        seedlings::Vector{Vector{Int64}},
+        spec_ID::Int64
+    )
+        #% GENERATE SEEDS TO BE DISPERSED---------------------------#
+        n_seeds = trunc(Int, (rand(Poisson(seed_production)) * 
+                    (1 - ldd_disp_frac)))
+        
+        #% CALCULATE CROWN WIDTH AND NEIGHBOURS---------------------#
+        #? This is a simplified allometric relationship 
+        #?from SORTIE-NZ
+        cw = (0.284 .* ((dbh .* 100.0) .^ 0.654))
```

Comment:
I think adding `.=` avoids some unnecessary pre-assignments. 

```suggestion
        cw .= (0.284 .* ((dbh .* 100.0) .^ 0.654))
```

Equivalently, there is the macro sugar:

```julia
@. cw = (0.284 * ((dbh * 100.0) ^ 0.654))
```